### ISSUE TITLE
VS2022 Razor editor - No way to get full completion list within @bind-Value

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#36347]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/VS2022-Razor-editor---No-way-to-get-full/1519516)._

---
[severity:It's more difficult to complete my work] [regression] [worked-in:15]
See the following screen capture. I want to get a completion list to show the available properties on the Person type. It is possible to do that outside the @bind-Value attribute, but seemingly not possible to do it inside, at least not without guessing the first character of the property name which doesn't help if you don't know all the property names.

Repro: Using the code from the screen capture, type "@person." on line 3 and see the completion list appear correctly. Then type the same thing withing the @bind-Value="" attribute on line 10, and see that the completion list fails to appear. Also you can't make it appear even by pressing ctrl+dot.

If you type the first letter of the property (e.g., "N") then the completion list does appear, but that's not helpful enough if you don't know the first letter of all the possible properties.

! [vs-razor-person-completion-2022.gif] (https://aka.ms/dc/image?name=Baf5bdcba06274404ab52e0f4d1122c0c637660965406256859_vs-razor-person-completion-2022.gif&tid=af5bdcba06274404ab52e0f4d1122c0c637660965406256859)

---
#### Feedback Bot on 9/1/2021, 11:01 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by SteveSandersonMS** [comment]
@NTaylorMullen Is this being moved here because you’re saying the problem is in the Razor compiler?

**Comment by NTaylorMullen** [comment]
> @NTaylorMullen Is this being moved here because you’re saying the problem is in the Razor compiler?

Nope, we still track Razor tooling issues in dotnet/aspnetcore under the area-razor.tooling label 😄

**Comment by NTaylorMullen** [comment]
@mkArtakMSFT I investigated and this is a compiler issue. Here's the gist given this example (trying to get completion after `this.`:
![image](https://user-images.githubusercontent.com/2008729/134741421-8216afb9-6f13-4355-9b28-721a00457d03.png)

- Compiler marks `this.` as "Code" in the syntax tree (we then rightfully think it's C#)
- Compiler does **not** generate C# source mappings for this location so tooling doesn't know where to re-invoke completion

Moving over to compiler.

Lol @SteveSandersonMS my initial intention was definitely not to move this to compiler but turned out that way? 😆
